OrbitBase/include/OrbitBase/Action.h

```diff
@@ -32,10 +34,30 @@ class NullaryFunctorAction : public Action {
   F functor_;
 };
 
+// This class implements an action that calls a method for some object.
+template <typename T>
+class MethodAction : public Action {
+ public:
+  explicit MethodAction(T* object,
+                        void (internal::identity<T>::type::*method)())
```

Comment:
Using T::*method wouldn't work here? What is the benefit of wrapping type T in internal::identity<T>::type? Also, would it be possible to extend to methods with arguments?  Maybe std::function and variadic generic lambdas (c++ 14) could help here?

Reply:
if you remove it compilation will fail - the test case specifically created derived class in order to test this. We already have a version with generic lambda, there is no need to have a variadic one because it parameters can be captured.